bricepollock/canyoneer
# Move to mapbox

### Description
Update to mapbox on main map from Apple Maps to get topo lines on map. This required doing some client-side finagling since the cloud mapbox provider engineer decided to discontinue work on this project. 

Detail Notes:
* Updated to non-backwards compatible Mapbox SDK
* Dropped idea we could hide map behind abstraction as that was more effort than payoff when Map APIs were considerably different
* Client side workaround not the most efficient but seems "good enough" on device
* Fixed a few minor bugs
* Fix Search incrementally coming in 

Issues:
Still not great in there is some issues with collisions between names and icons even though that shouldn't be allowed according to SDK.

### Test Plan
* View canyons on map
* See favorite upgrade
* View single canyon
* Search
* View canyon details
